4 - AnimalShelter/Code/AnimalShelter.cs

```diff
@@ -0,0 +1,81 @@
+﻿using AnimalShelterCode;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace AnimalShelterCode
+{
+    public class AnimalShelter
+    {
+        public int AnimalCounter { get; private set; }
+
+        public List<Animal> Animals { get; private set; }
+
+        public AnimalShelter()
+        {
+            Animals = new List<Animal>();
+            AnimalCounter = 1;
+        }
+
+        public OperationResult AddAnimal(Animal animal)
+        {
+            if (animal.Type == AnimalType.Bear || animal.Type == AnimalType.Deer) 
+            { 
+                return new OperationResult($"Animal Type {animal.Type} is not Supported");
+            }
```

Comment:
Doing checks for "not supported" does totally work, but we warned its normally something we don't want to do unless we need to. We often should check for "types supported" instead because what if in the future the animal type needs to be expanded and a new type is added? Your logic would now break and chances are your unit tests won't catch this.